src/TextField/TextField.js

```diff
@@ -119,6 +122,26 @@ export class TextField extends React.PureComponent {
     this._validate(e, value, false);
   };
 
+  _handleClick = e => {
+    e.persist();
+    if (typeof this.props.onClick !== "undefined") {
```

Comment:
Prop types protects this from not being a function, but this could be `typeof === "function"` since we're calling it as a function. No real preference, but without prop types it would a type error.